### ISSUE TITLE
Feature request: AMP support

### ISSUE THREAD

**Issue by slavino** [ikkez/f3-mailer#4]
https://www.ampproject.org/docs/integration/integrate-amphtml-email

#1 stage 1
just add setAMP() method

#2 stage 2
tracking options to be investigated (if needed as the part is interactive enough)

**Comment by ikkez** [comment]
Hi. Not sure what you want from me here. As far as I can see, you want me to add a frontend system that implements a "viewer / player" for inside html emails, which should someday display even more html mail stuff than today, that is currently under construction and no mail provider or client does even have support for it build in yet. So.. are you serious? 😆

**Comment by slavino** [comment]
You may have missed it, so let me google it for you.

https://ampbyexample.com/playground/#runtime=amp4email

https://www.theverge.com/2019/3/26/18282636/amp-gmail-shopping-rsvp-forms-inbox-google-beta

https://www.emailmarketingtipps.de/2018/02/19/amp-for-email-controversies/

**Comment by ikkez** [comment]
ok yeah fine.. not sure if i should like it or not.. but at least, it's just  html template, with some specialized markup - so nothing an SMTP mailer should care about. Your AMP support is there, if you integrate it into your template.

**Comment by slavino** [comment]
Maybe yet another example will help to understand? :)

https://www.ampproject.org/docs/interaction_dynamic/amp-email-format

```
From:  Person A <persona@gmail.com>
To: Person B <personb@gmail.com>
Subject: An AMP email!
Content-Type: multipart/alternative; boundary="001a114634ac3555ae05525685ae"

--001a114634ac3555ae05525685ae
Content-Type: text/plain; charset="UTF-8"; format=flowed; delsp=yes

Hello World in plain text!

--001a114634ac3555ae05525685ae
Content-Type: text/x-amp-html; charset="UTF-8"

<!doctype html>
<html ⚡4email>
<head>
  <meta charset="utf-8">
  <style amp4email-boilerplate>body{visibility:hidden}</style>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
</head>
<body>
Hello World in AMP!
</body>
</html>
--001a114634ac3555ae05525685ae--
Content-Type: text/html; charset="UTF-8"

<span>Hello World in HTML!</span>
--001a114634ac3555ae05525685ae
```

**Comment by ikkez** [comment]
Implemented in latest release v1.2.0 :wink:

```php
$mailer = new Mailer();
$mailer->addTo('mail@domain.com');
$mailer->setText('Hello World');

$amp = '<!doctype html>
<html ⚡4email>
<head>
  <meta charset="utf-8">
  <style amp4email-boilerplate>body{visibility:hidden}</style>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
</head>
<body>
Hello World in AMP!
</body>
</html>';
$mailer->setContent($amp,'text/x-amp-html');

$mailer->send('AMP Testmail');
```

You need to whitelist your sender email for testing in Gmail though: https://developers.google.com/gmail/ampemail/testing-dynamic-email
